### PR TITLE
Expose HostedCluster and NodePool metrics in HyperShift operator

### PR DESCRIPTION
**Pull request by csrwng**
**What this PR does / why we need it**:
Generates metrics to answer questions like:
- How many hosted clusters are there in a management clusters? what
  platform do they use? what OCP version are they using?
- How many NodePools are there in a management cluster? are they using
  autorepair, autoscaling? what OCP version are they using?
- Are the HostedClusters/NodePools healthy? Do they have any conditions
  in an undesirable state?
- How many NodePools are associated with a HostedCluster?
- How many replicas are there per NodePool?

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes # HOSTEDCP-401

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.

### CODE REVIEW

File: hypershift-operator/metrics.go [hunk 1]
```diff
@@ -20,11 +23,53 @@ type hypershiftMetrics struct {
...
+			Name: "hypershift_cluster_initial_rollout_duration_seconds",
+		}, []string{"name"}),
+		hostedClusters: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Name: "hypershift_hostedclusters",
+			Help: "Total number of HostedClusters by platform, endpoint_access, version",
+		}, []string{"platform", "endpoint_access", "version"}),
+		hostedClustersWithCondition: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Name: "hypershift_hostedclusters_conditions",
+			Help: "Total number of HostedClusters by platform with conditions in undesired state",
+		}, []string{"platform", "condition"}),
```
Comment (alvaroaleman) [inline hypershift-operator/metrics.go:51]: Maybe name it `failure_conditions` or sth? The help text is not visible in Prom and that label name is not particularly self-explanatory

File: hypershift-operator/metrics.go [hunk 2]
```diff
@@ -20,11 +23,53 @@ type hypershiftMetrics struct {
...
+		hostedClustersNodePools: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Name: "hypershift_hostedclusters_nodepool_count",
+			Help: "Number of NodePools associated with a given HostedCluster",
+		}, []string{"name", "platform"}),
```
Comment (alvaroaleman) [inline hypershift-operator/metrics.go:55]: maybe name this `cluster_name` instead to clarify its not the nodepool name?

**Comment by csrwng** [comment]
Thanks @alvaroaleman, updated the code

**Pull request merged**